z2/src/converter.rs

```diff
@@ -206,13 +208,55 @@ fn get_contract_code(zq1_db: &zq1::Db, address: Address) -> Result<Code> {
         })?
     };
 
-    Ok(Code::Scilla {
-        code: String::from_utf8(code)
-            .map_err(|err| anyhow!("Unable to convert scilla code into string: {err}"))?,
+    let code = String::from_utf8(code)
+        .map_err(|err| anyhow!("Unable to convert scilla code into string: {err}"))?;
+
+    let contract = Contract::from_str(&code)?;
+
+    let scilla_code = Code::Scilla {
+        code: code.clone(),
         init_data: init_data_vec,
         types: BTreeMap::default(),
-        transitions: vec![],
-    })
+        transitions: contract
+            .transitions
```

Comment:
Elsewhere in ZQ2 we rely on the Scilla interpreter's check output to calculate the `transitions` for a contract. Whereas here we are using the Rust parser. While these parsers should be the same in theory, it worries me a bit that they will differ for some edge cases. I would prefer if we used the Scilla interpreter here as well if possible.

Or if that's not possible, can we add a test suite to the Rust Scilla parser which takes every contract from testnet and mainnet and asserts that the parsed result is the same as what the OCaml implementation returns?